oaknational/Oak-Web-Application
# fix(LESQ-1155): guard against duplicate sign-up events when hook is remounted

## Description

Music year: 1953

## Issue(s)

Fixes
- A race condition causing sign-up events to be tracked multiple times

## How to test

1. [With the `teacher-download-auth` feature flag set to `with-login` ](https://eu.posthog.com/project/124/feature_flags/35482) and while signed-out
2. Go to https://deploy-preview-2953--oak-web-application.netlify.thenational.academy/teachers/programmes/computing-secondary-ks3-l/units/impact-of-technology-collaborating-online-respectfully-35d0/lessons/account-security-68rkee
4. Click "Sign in to download"
5. With the dev console opened throttle your connection to "Fast 4G" (or sit near a source of interference to disrupt your WiFi a bit 🤸🏻‍♂️ maybe)
6. Complete the sign-up process
7. When you're redirected to `/onboarding` you should see only one "User Sign-Up Completed" event sent to Posthog
8. This is a race condition so this might need to be repeated several times to be sure (though I was fairly reliably able to reproduce the condition with the throttled connection)

## Checklist
